src/routes.rs

```diff
@@ -222,15 +246,12 @@ pub async fn dispense_tokens(
         )
         .await
         .map(|r| r.map_err(|e| error(format!("Failed to submit transaction: {}", e))))
-        .map_err(|e| {
-            error(format!(
-                "Timeout during while submitting transaction: {}",
-                e
-            ))
-        });
+        .map_err(|e| error(format!("Timeout while submitting transaction: {}", e)));
 
         match result {
             Ok(Ok(_)) => {
+                dispense_tracker.lock().unwrap().track(address);
```

Comment:
Did you use `unwrap` here because this `lock` call guaranteed to succeed?

Reply:
There is no fallible operations being done while the lock is held, so assumed we are safe to unwrap